proposals/GlobalUsingDirective.md

```diff
@@ -2,13 +2,194 @@
+            * If the location where the *namespace_or_type_name* occurs is enclosed by a namespace declaration for `N` and the namespace declaration contains an *extern_alias_directive* or *using_alias_directive* that associates the name `I` with a namespace or type, **or any namespace declaration for `N` in the program contains a *global_using_alias_directive* that associates the name `I` with a namespace or type,** then the *namespace_or_type_name* is ambiguous and a compile-time error occurs.
+            * Otherwise, the *namespace_or_type_name* refers to the namespace named `I` in `N`.
+        * Otherwise, if `N` contains an accessible type having name `I` and `K` type parameters, then:
+            * If `K` is zero and the location where the *namespace_or_type_name* occurs is enclosed by a namespace declaration for `N` and the namespace declaration contains an *extern_alias_directive* or *using_alias_directive* that associates the name `I` with a namespace or type, **or any namespace declaration for `N` in the program contains a *global_using_alias_directive* that associates the name `I` with a namespace or type,** then the *namespace_or_type_name* is ambiguous and a compile-time error occurs.
+            * Otherwise, the *namespace_or_type_name* refers to the type constructed with the given type arguments.
+        * Otherwise, if the location where the *namespace_or_type_name* occurs is enclosed by a namespace declaration for `N`:
+            * If `K` is zero and the namespace declaration contains an *extern_alias_directive* or *using_alias_directive* that associates the name `I` with an imported namespace or type, **or any namespace declaration for `N` in the program contains a *global_using_alias_directive* that associates the name `I` with an imported namespace or type,** then the *namespace_or_type_name* refers to that namespace or type.
+            * Otherwise, if the namespaces and type declarations imported by the *using_namespace_directive*s and *using_alias_directive*s of the namespace declaration contain exactly one accessible type having name `I` and `K` type parameters, then the *namespace_or_type_name* refers to that type constructed with the given type arguments.
+            * Otherwise, if the namespaces and type declarations imported by the *using_namespace_directive*s and *using_alias_directive*s of the namespace declaration contain more than one accessible type having name `I` and `K` type parameters, then the *namespace_or_type_name* is ambiguous and an error occurs.
+            * **Otherwise, if the namespaces and type declarations imported by the *global_using_namespace_directive*s and *global_using_static_directive*s of any namespace declaration for `N` in the program contain exactly one accessible type having name `I` and `K` type parameters, then the *namespace_or_type_name* refers to that type constructed with the given type arguments.**
```

Comment:
The wording of this is odd. I would expect that if a global using adds one definition of `A` and a regular using adds a different definition of `A`, then lookup would be ambiguous.

Reply:
> I would expect that if a global using adds one definition of A and a regular using adds a different definition of A, then lookup would be ambiguous.

That is an option, but I am proposing a different approach. We will discuss this at LDM and make the decision. 

---
In reply to: [589639148](https://github.com/dotnet/csharplang/pull/4499#discussion_r589639148) [](ancestors = 589639148)

Reply:
> We will discuss this at LDM and make the decision. 

Please consider adding a note that it's an open question then.

---
In reply to: [589656318](https://github.com/dotnet/csharplang/pull/4499#discussion_r589656318) [](ancestors = 589656318,589639148)